app.rb

```diff
@@ -73,14 +71,15 @@ def store_record decoded
     return [decoded['id'], 'ERROR']
   end
 
-  $logger.debug "Response", { "resp": JSON.parse(resp.body)}
-
-  $logger.info "Bib #{decoded['nyplSource']} #{decoded['id']} successfully processed by Subject Heading (SHEP) API" if resp.code.to_i == 201
-  return [decoded['id'], 'SUCCESS']
-end
+  $logger.debug 'Response', { code: resp.code.to_i, resp: resp.body }
 
-def should_process? data
-  is_research?(data) && have_subject_headings_changed?(data)
+  if resp.code.to_i == 304
+    $logger.info "Bib #{decoded['nyplSource']} #{decoded['id']} had no modifications for Subject Heading (SHEP) API"
+    return [decoded['id'], 'NOT MODIFIED']
+  elsif resp.code.to_i == 201
+    $logger.info "Bib #{decoded['nyplSource']} #{decoded['id']} successfully processed by Subject Heading (SHEP) API"
+    return [decoded['id'], 'SUCCESS']
+  end
```

Comment:
Should there be a base case that handles unexpected HTTP codes? There is a test that says to expect `nil` if one is encountered (e.g. 200) but does the method that invokes `process_record` know how to handle that?

Reply:
Good question. I wondered about that myself, but as far as I can see it doesn't do anything with it, so it should make no difference.

Reply:
yeah, I wonder if this function isn't doing anything with these return values maybe we should just return nil in all of these instances? They all get loaded into a list on line 30, maybe we should at least log that? Or if not we should just return `nil` in all cases so there isn't future confusion about the utility of these returns.

Reply:
The one thing it is useful for is to enable clear tests. Which isn't the best reader, but probably easier than mocking the `$logger`. I agree we shouldn't store the results, that is confusing. Updating that now.

Reply:
looks good to me! good point about testing (and there is a very good chance I was the one who added that line haha, always fun to come back to code you wrote a while ago)